src/features/session_replay/aggregate/index.js

```diff
@@ -226,15 +235,15 @@ export class Aggregate extends AggregateBase {
 
     // do not let the last node be a meta node, since this NEEDS to precede a snapshot
     // we will manually inject it later if we find a payload that is missing a meta node
-    const payloadEndsWithMeta = this.events[this.events.length - 1]?.type === 4
+    const payloadEndsWithMeta = this.events[this.events.length - 1]?.type === RRWEB_EVENT_TYPES.Meta
     if (payloadEndsWithMeta) {
-      this.hasMeta = false
       this.events = this.events.slice(0, this.events.length - 1)
+      this.hasMeta = !!this.events.find(x => x.type === RRWEB_EVENT_TYPES.Meta)
     }
 
     // do not let the first node be a snapshot node, since this NEEDS to be preceded by a meta node
     // we will manually inject it if this happens
-    const payloadStartsWithSnapshot = this.events[0]?.type === 4
+    const payloadStartsWithSnapshot = this.events[0]?.type === RRWEB_EVENT_TYPES.FullSnapshot
```

Comment:
Maybe change this variable to `payloadStartsWithFullSnapshot`. It's more clear that we don't care about partial snapshot nodes and variable name length doesn't really matter.